File: android/src/main/java/com/fiwind/plugins/capacitor/pushprovisioning/PushProvisioningCapacitorPluginPlugin.java
package com.fiwind.plugins.capacitor.pushprovisioning;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "PushProvisioningCapacitorPlugin")
public class PushProvisioningCapacitorPluginPlugin extends Plugin {

    @PluginMethod
    public void isAvailable(PluginCall call) {
        // FIXME: Not implemented
        JSObject ret = new JSObject();
        ret.put("available", true);
        call.resolve(ret);
    }

    @PluginMethod
    public void isPaired(PluginCall call) {
        // FIXME: Not implemented
        JSObject ret = new JSObject();
        ret.put("paired", false);
        call.resolve(ret);
    }

    @PluginMethod
    public void getCardUrl(PluginCall call) {
        // FIXME: Not implemented
        JSObject ret = new JSObject();
        ret.put("url", null);
        call.resolve(ret);
    }

    @PluginMethod
    public void startEnroll(PluginCall call) {
        call.reject("startEnroll() is not implemented yet");
    }

    @PluginMethod
    public void completeEnroll(PluginCall call) {
        call.reject("completeEnroll() is not implemented yet");
    }
}

File: ios/Plugin/PushProvisioningCapacitorPluginPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(PushProvisioningCapacitorPluginPlugin, "PushProvisioningCapacitorPlugin",
           CAP_PLUGIN_METHOD(isAvailable, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(isPaired, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(getCardUrl, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(startEnroll, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(completeEnroll, CAPPluginReturnPromise);
)
